src/ray/gcs/gcs_server/actor_info_handler_impl.cc

```diff
@@ -0,0 +1,84 @@
+#include "actor_info_handler_impl.h"
+#include "ray/util/logging.h"
+
+namespace ray {
+namespace rpc {
+
+void DefaultActorInfoHandler::HandleGetActorInfo(
+    const rpc::GetActorInfoRequest &request, rpc::GetActorInfoReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  RAY_LOG(DEBUG) << "Begin get actor info, actor id is:" << request.actor_id();
+  ActorID actor_id = ActorID::FromBinary(request.actor_id());
+  auto on_done = [request, reply, send_reply_callback](
+                     Status status, const boost::optional<ActorTableData> &result) {
+    if (status.ok()) {
+      if (result) {
+        reply->mutable_actor_table_data()->CopyFrom(*result);
+      }
+    } else {
+      RAY_LOG(ERROR) << "Failed to get actor info:" << status.ToString()
+                     << ",actor id is:" << request.actor_id();
+    }
+    send_reply_callback(status, nullptr, nullptr);
+  };
+
+  Status status = gcs_client_.Actors().AsyncGet(actor_id, on_done);
+  if (!status.ok()) {
+    on_done(status, boost::none);
+  }
+  RAY_LOG(DEBUG) << "Finish get actor info, actor id is:" << request.actor_id();
+}
+
+void DefaultActorInfoHandler::HandleRegisterActorInfo(
+    const rpc::RegisterActorInfoRequest &request, rpc::RegisterActorInfoReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  RAY_LOG(DEBUG) << "Begin register actor info, actor id is:"
+                 << request.actor_table_data().actor_id();
+  auto actor_table_data = std::make_shared<ActorTableData>();
+  actor_table_data->CopyFrom(request.actor_table_data());
+  auto on_done = [request, send_reply_callback](Status status) {
+    if (!status.ok()) {
+      RAY_LOG(ERROR) << "Failed to register actor info:" << status.ToString()
+                     << ",actor id is:" << request.actor_table_data().actor_id();
+    }
+    send_reply_callback(status, nullptr, nullptr);
+  };
+
+  Status status = gcs_client_.Actors().AsyncRegister(
+      actor_table_data, [send_reply_callback](Status status) {
+        send_reply_callback(status, nullptr, nullptr);
+      });
+  if (!status.ok()) {
+    on_done(status);
```

Comment:
This line is not consistent with L49, should call the same func at successful and failed cases

Reply:
both places should call on_done?

Reply:
when status is not ok, would it be possible for the callback to be invoked twice?

Reply:
If status is not ok, redis callback will not be invoked because redis server do not received request, for example: redis command is invalid.If redis server received request, status is ok and redis callback will be invoked.

Reply:
could you set the callback to be consistent in the two cases? same for places below

Reply:
Extract on_done function is to reduce code redundancy, for example：
RAY_LOG(ERROR) << "Failed to register actor info: " << status.ToString() << ", actor id = " << actor_id;  and send_reply_callback(status, nullptr, nullptr);  only needs one time.